### PR TITLE
Feat/condo/doma 7455/marketplace settings pages

### PR DESCRIPTION
**Pull request by abshnko**
Flow to setup marketplace -- WIP

Screenshot 2023-11-02 at 00 42 13
Screenshot 2023-11-02 at 00 42 35
Screenshot 2023-11-02 at 00 42 56
Screenshot 2023-11-02 at 00 44 02

### CODE REVIEW

File: apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    const organizationId = get(launchContext, 'condoContextEntityId', null)
+    const userId = get(launchContext, 'condoUserId', null)
+    const createAcceptAction = Accept.useCreate({})
+    const integrationId = get(integrationConfig, 'id')
+    //TODO: can't get context -- rights?
+    const {
+        obj: context,
+        loading: contextIsLoading,
+        refetch: refetchContext,
+    } = AcquiringIntegrationContext.useObject(
```
Comment (VKislov) [inline apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx:72]: InvoiceContext

File: apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetupPage.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    const { organization } = useOrganization()
+    const orgId = get(organization, 'id', null)
+    const orgType = get(organization, 'type', MANAGING_COMPANY_TYPE)
+
+    const createAction = AcquiringContext.useCreate({
+        status: CONTEXT_IN_PROGRESS_STATUS,
+        settings: { dv: 1 },
+        state: { dv: 1 },
+    })
+    const updateAction = AcquiringContext.useUpdate({
```
Comment (VKislov) [inline apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetupPage.tsx:32]: there is no need to update the status, this is done in afterChange of the Accept scheme

File: apps/condo/domains/marketplace/components/MarketplaceOnboarding/RequisitesSetup.tsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            },
+            taxRegime: values.taxType,
+            vatPercent: String(values.taxPercent),
+            currencyCode: 'RUB',
+        }).then(()=>{
+            router.replace({ query: { step: 1 } })
+        })
+
+        setIsLoading(false)
+    }, [intl])
```
Comment (VKislov) [inline apps/condo/domains/marketplace/components/MarketplaceOnboarding/RequisitesSetup.tsx:127]: why intl? Perhaps `organization` is also needed here, but I'm not sure

File: apps/condo/domains/marketplace/components/MarketplaceOnboarding/RequisitesSetup.tsx [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+                                        ) : <Input/>}
+                                    </Col>
+                                </Form.Item>
+                            </Col>
+                            <Col span={24}>
+                                <Form.Item
+                                    label={TaxTypeLabel}
+                                    name='taxType'
+                                    required
+                                    labelCol={{ lg: 10 }}
```
Comment (VKislov) [inline apps/condo/domains/marketplace/components/MarketplaceOnboarding/RequisitesSetup.tsx:212]: you can put this into a constant, but this is optional :)

**Review by VKislov — APPROVED** [review]

File: apps/condo/domains/marketplace/components/MarketplacePageContent/ContextProvider.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext } from 'react'
+
+import type { InvoiceContext as InvoiceContextSchema } from '@app/condo/schema'
+
+type IInvoiceContext = {
+    invoiceContext: InvoiceContextSchema
+    refetchInvoice: () => void
```
Comment (AleX83Xpert) [inline apps/condo/domains/marketplace/components/MarketplacePageContent/ContextProvider.ts:7]: Context?
```suggestion
    refetchInvoiceContext: () => void
```

File: apps/condo/domains/marketplace/components/MarketplacePageContent/ContextProvider.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export const InvoiceContext = createContext<IInvoiceContext>({
+    invoiceContext: null,
+    refetchInvoice: () => ({}),
+})
+
+export function useBillingAndAcquiringContexts (): IInvoiceContext {
```
Comment (AleX83Xpert) [inline apps/condo/domains/marketplace/components/MarketplacePageContent/ContextProvider.ts:15]: Billing and acquiring?

File: apps/condo/domains/marketplace/schema/InvoiceContext.js [hunk 7]
```diff
@@ -127,6 +128,37 @@ const InvoiceContext = new GQLListSchema('InvoiceContext', {
...
+        resolveInput: async ({ operation, resolvedData, context }) => {
+            if (operation === 'create') {
+                const { recipient } = resolvedData
+                if (!recipient) {
+                    return resolvedData
+                }
+
+                const { tin, bic } = recipient
+                if (!tin || !bic) {
+                    throw new Error('No tin or bic passed')
```
Comment (AleX83Xpert) [inline apps/condo/domains/marketplace/schema/InvoiceContext.js:140]: What about GQLError + test?

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+import Input from '@condo/domains/common/components/antd/Input'
+import { FormWithAction } from '@condo/domains/common/components/containers/FormList'
+import { Loader } from '@condo/domains/common/components/Loader'
+import { useValidations } from '@condo/domains/common/hooks/useValidations'
+import {
+    MARKETPLACE_RULES_LINK,
+    MARKETPLACE_OFFER_LINK,
+} from '@condo/domains/marketplace/constants'
+import { Organization } from '@condorb/domains/condo/utils/clientSchema'
```
Comment (pahaz) [inline apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx:19]: !! condo !!

File: apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+import { SCOPE_TYPES } from '@condorb/domains/condorb/constants/marketplace'
```
Comment (pahaz) [inline apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx:20]: !! condo !!

File: apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+import { Accept } from '@condorb/domains/condorb/utils/clientSchema'
```
Comment (pahaz) [inline apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx:21]: !! condo !!

File: apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx (hunk 8 above)
Comment (AleX83Xpert) [reply apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx:19]: fixed

File: apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx (hunk 9 above)
Comment (AleX83Xpert) [reply apps/condo/domains/marketplace/components/MarketplaceOnboarding/OfferSetup.tsx:20]: fixed

**Pull request merged**